app/src/main/java/com/alexstyl/themingandroid/AppCompatThemedActivity.java

```diff
@@ -38,10 +38,10 @@ protected final void applyTheme(AppTheme theme) {
     }
 
     private void restartActivity() {
-        Intent intent = getIntent();
+        final Intent intent = getIntent();
         intent.removeCategory(Intent.CATEGORY_LAUNCHER);
-        startActivity(intent);
         finish();
+        startActivity(intent);
```

Comment:
What is the reasoning behind finishing the activity before starting the new one?